sdk/servicebus/service-bus/src/serviceBusMessage.ts

```diff
@@ -572,18 +577,20 @@ export function fromRheaMessage(
       props.lockedUntilUtc = new Date(msg.message_annotations[Constants.lockedUntil] as number);
     }
   }
-  if (msg.ttl != null && msg.ttl >= Constants.maxDurationValue - props.enqueuedTimeUtc.getTime()) {
-    props.expiresAtUtc = new Date(Constants.maxDurationValue);
-  } else {
-    props.expiresAtUtc = new Date(props.enqueuedTimeUtc.getTime() + msg.ttl!);
+  if (msg.ttl == null) msg.ttl = Constants.maxDurationValue;
+  if (props.enqueuedTimeUtc) {
+    if (msg.ttl >= Constants.maxDurationValue - props.enqueuedTimeUtc.getTime()) {
```

Comment:
(is this the same?)
props.expiresAtUtc = Math.min(props.enqueuedTimeUtc.getTime() + msg.ttl, Constants.maxDurationValue)

Reply:
Great catch

Reply:
>     props.expiresAtUtc = new Date(
      Math.min(props.enqueuedTimeUtc.getTime() + msg.ttl, Constants.maxDurationValue)
    );